src/Simulation/QsharpCore/Intrinsic.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Intrinsic
+{
+    public partial class CNOT
+    {
+        /// <inheritdoc/>
+        public override RuntimeMetadata? GetRuntimeMetadata(IApplyData args)
+        {
+            if (args.Value is ValueTuple<Qubit, Qubit> cnotArgs)
+            {
+                var (ctrl, target) = cnotArgs;
+                return new RuntimeMetadata()
+                {
+                    Label = "X",
+                    IsControlled = true,
+                    Controls = new List<Qubit>() { ctrl },
+                    Targets = new List<Qubit>() { target },
+                };
+            }
+            else
+            {
+                Console.WriteLine($"Failed to retrieve runtime metadata for {this.ToString()}.");
+                return null;
+            }
+        }
+    }
+
+    public partial class CCNOT
+    {
+        /// <inheritdoc/>
+        public override RuntimeMetadata? GetRuntimeMetadata(IApplyData args)
+        {
+            if (args.Value is ValueTuple<Qubit, Qubit, Qubit> ccnotArgs)
+            {
+                var (ctrl1, ctrl2, target) = ccnotArgs;
+                return new RuntimeMetadata()
+                {
+                    Label = "X",
+                    IsControlled = true,
+                    Controls = new List<Qubit>() { ctrl1, ctrl2 },
+                    Targets = new List<Qubit>() { target },
+                };
+            }
+            else
+            {
+                Console.WriteLine($"Failed to retrieve runtime metadata for {this.ToString()}.");
+                return null;
+            }
+        }
+    }
+
+    public partial class M
+    {
+        /// <inheritdoc/>
+        public override RuntimeMetadata? GetRuntimeMetadata(IApplyData args)
+        {
+            if (args.Value is Qubit target)
+            {
+                return new RuntimeMetadata()
+                {
+                    Label = ((ICallable)this).Name,
+                    IsMeasurement = true,
+                    Targets = new List<Qubit>() { target },
+                };
+            }
+            else
+            {
+                Console.WriteLine($"Failed to retrieve runtime metadata for {this.ToString()}.");
+                return null;
+            }            
+        }
+    }
+
+    public partial class Reset
+    {
+        /// <inheritdoc/>
+        public override RuntimeMetadata? GetRuntimeMetadata(IApplyData args) => null;
+    }
+
+    public partial class ResetAll
+    {
+        /// <inheritdoc/>
+        public override RuntimeMetadata? GetRuntimeMetadata(IApplyData args) => null;
+    }
```

Comment:
Is there a reason these two don't have associated metadata?

Reply:
These operations aren't used in the visualization, so I return null to indicate that I don't intend on processing these operations. However, if there is another use for `RuntimeMetadata` in the future, it might make more sense to include a `Display=false` attribute or something of that flavour.

Reply:
I'd suggest that including resets in visualization would make a lot of sense; resetting a qubit or not can significantly change the meaning of a quantum program.

Reply:
I see, would it make sense to implement this in a separate PR? cc-ing @anpaz-msft.

Reply:
We decided to remove this override and render it as a default gate visualization.